src/UserWidget.js

```diff
@@ -346,5 +317,67 @@ var SystemButton = GObject.registerClass(
                 this.notify('active');
             }
         }
+    });
+
+var DoNotDisturbSwitch = GObject.registerClass({
+    },
+    class DoNotDisturbSwitch extends St.Button {
+        _init(iconSize,useIcon) {
+            super._init({
+                style_class: 'bttn system-button dnd-button',
+                can_focus: true,
+                toggle_mode: true,
+                y_align: Clutter.ActorAlign.CENTER
+            });
+
+            this._settings = new Gio.Settings({
+                schema_id: 'org.gnome.desktop.notifications',
+            });
+
+            this._show_banners = this._settings.get_boolean('show-banners');
+
+            if (useIcon){
+                this._icon = new St.Icon ({iconSize: iconSize});
+                this.setIcon(this._show_banners);
+
+                this._settings.bind('show-banners',
+                                    this, 'checked',
+                                    Gio.SettingsBindFlags.DEFAULT);
+
+                this.connect ('clicked', () => {
+                    this._settings.set_boolean('show-banners',!(this._show_banners));
+
+                    this._show_banners = this._settings.get_boolean('show-banners');
+
+                    this.setIcon(this._show_banners);
+                });
+
+                this.set_child(this._icon);
+            } else {
+                this._switch = new PopupMenu.Switch(this._show_banners);
+
+                this._settings.bind('show-banners',
+                                    this._switch, 'state',
+                                    Gio.SettingsBindFlags.INVERT_BOOLEAN);
```

Comment:
I think this flag shouldn't be inverted. I think it is better when the switch is checked and notifications active. 

```
this._settings.bind('show-banners',
                                    this._switch, 'state',
                                    Gio.SettingsBindFlags.BOOLEAN);
```

Reply:
That is how it is actually in default switch in calendar since it is actually turn on do not disturb mode ( turn off notifications ) instead of switch notifications, but will fix it to regular.

Reply:
Yes, I know that but I've changed this :) and it is better with BOOLEAN not inverted